src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.commands;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.task.Status;
+import seedu.address.model.task.Task;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class MarkCommand extends Command {
+    public static final String COMMAND_WORD = "mark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Marks the task identified by the index number used in the displayed task list as completed.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_MARK_TASK_SUCCESS = "Task Completed: %1$s";
+    public static final String MESSAGE_MARK_MARKED_TASK = "The task is already marked";
+
+    private final Index targetIndex;
+
+    public MarkCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Task> lastShownList = model.getFilteredTaskList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        Task taskToMark = lastShownList.get(targetIndex.getZeroBased());
+
+        Status status = taskToMark.getStatus();
+        if (status.isCompleted()) {
+            throw new CommandException(MESSAGE_MARK_MARKED_TASK);
+        }
+
+        Task markedTask = new Task(taskToMark.getDescription(), status.updateStatus());
+
+        // to change
```

Comment:
Could you explain what `to change` mean in this context?
If there are things to add on in the future i thing use the following formatting for syntax highlighting
```
// TODO: what needs to be done
```

Reply:
It was a personal note to myself to properly implement the specific changes, I apologise about it haha
Will personal notes to remove in future PRs!